test/opml.cpp

```diff
@@ -0,0 +1,312 @@
+#include "opml.h"
+
+#include <libxml/xmlsave.h>
+#include "cache.h"
+#include "file_urlreader.h"
+
+#include "3rd-party/catch.hpp"
+#include "test-helpers.h"
+
+using namespace newsboat;
+
+TEST_CASE("prepare_opml creates an XML document with feed URLs in OPML format",
+		"[opml]")
+{
+	const auto check =
+		[]
+		(const FeedContainer& feedcontainer)
+		-> std::string
```

Comment:
Does clang-format format it like that? feels like a waste. 3 lines that could be one and wouldn't lose readability.